drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -321,6 +321,62 @@ static int bq27427_ccgain_quirk(const struct device *dev)
 	return 0;
 }
 
+static int bq274xx_ensure_chemistry(const struct device *dev, uint16_t chem_id)
+{
+	struct bq274xx_data *data = dev->data;
+	int ret;
+	uint16_t val;
+
+	ret = bq274xx_ctrl_reg_write(dev, BQ274XX_CTRL_CHEM_ID);
+	if (ret < 0) {
+		LOG_ERR("Unable to write control register");
+		return -EIO;
+	}
+
+	ret = bq274xx_cmd_reg_read(dev, BQ274XX_CMD_CONTROL, &val);
+	if (ret < 0) {
+		LOG_ERR("Unable to read register");
+		return -EIO;
+	}
+
+	LOG_DBG("Chem ID: %04x", val);
+
+	if (val != chem_id) {
+		/* Only the bq27427 has a configurable Chemistry ID. On bq27421, it depends on the
+		 * variant of the chip, so just error out if the chemistry ID is wrong.
+		 */
+		if (data->regs != &bq27427_regs) {
+			LOG_ERR("Unable to confirm chemistry ID 0x%04x. Device reported 0x%04x",
+				chem_id, val);
+			return -EIO;
+		}
+
+		uint16_t cmd;
+
+		switch (val) {
+		case 0x3230:
+			cmd = 0x0030;
+			break;
+		case 0x1202:
+			cmd = 0x0031;
+			break;
+		case 0x3142:
+			cmd = 0x0032;
+			break;
```

Comment:
this can use a bunch of `#define`, how about:

- add defines for the cmd values (`BQ274XX_CHEM_A_CMD`... or something like that, in `bq274xx.h`)
- add a set of defines that can be used in the device tree for the value, those would have to go in a new `dt-bindings/sensor/bq274xx.h`, see the other files there